api/proto/auth/client.go

```diff
@@ -0,0 +1,186 @@
+package auth
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sync"
+	"sync/atomic"
+
+	"github.com/gravitational/roundtrip"
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+var log = logrus.WithFields(logrus.Fields{
+	trace.Component: teleport.ComponentAuth,
+})
+
+// Client is HTTP Auth API client. It works by connecting to auth servers
+// via HTTP.
+//
+// When Teleport servers connect to auth API, they usually establish an SSH
+// tunnel first, and then do HTTP-over-SSH. This client is wrapped by TunClient
+// in lib/auth/tun.go
+type Client struct {
+	sync.Mutex
+	ClientConfig
```

Comment:
I would import these explicitly. Put the mutex close to the thing it protects.

Reply:
What do you mean by importing explicitly?